JAVASCRIPT/Session_5/Extract_Nested_Object_Reference.js

```diff
@@ -0,0 +1,16 @@
+// Extract Nested Object Reference
+// http://www.codewars.com/kata/extract-nested-object-reference
+
+Object.prototype.hash = function(string) {
+    var prop_list = string.split('.');
+    var obj = this;
+
+    for (i = 0; i < prop_list.length; i++) {
+        if (obj[prop_list[i]] != undefined) {
```

Comment:
undefined check is simpliefied, 

if(something) is same as if(something!==undefined)